src/MergeContext.js

```diff
@@ -489,6 +489,8 @@ class PullRequest {
 
         this._updated = false; // _update() has been called
 
+        this._abandoned = false; // abandonedStagingChecksLabel() has been set
```

Comment:
It is not clear why we could not ask this._labels whether they have that specific label set. If we do need this data member, then the description needs adjustment, but I cannot suggest anything specific yet because of the other code oddities in the PR diff.